### DISCUSSION TITLE
Wavelog install in docker - Stuck at "Running database migrations"

### DISCUSSION THREAD

**Discussion by winnieXY** [wavelog/wavelog#680] [Q&A] [answered]
Hey,

I've set up wavelog on my server and tried two different ways to install it:

First I modified your provided docker-compose.yml so that it matches my setup (webthings always behind a proxy e.g.) which seems to work well initially but then the installation got stuck at "Running database migrations", and after 300s got a timeout from my proxy. I increased the proxypass timeout to 14400 seconds  - restarted again and now the timeout is obviously gone, but the circle is still spinning (a hour later).

Here is the log:
```
                           Wavelog Installer Debug Log
-------
Log Location: ../application/logs/installer_debug.log

INFO - 2024-08-02 07:21:29 --> Called Installer index.php
INFO - 2024-08-02 07:21:29 --> With URL: http://my-url.de:8086/install//
INFO - 2024-08-02 07:21:29 --> From IP: some-ipv6-ip
INFO - 2024-08-02 07:21:29 --> Set language cookie to "English"
DEBUG - 2024-08-02 07:23:43 --> Start writing config.php
DEBUG - 2024-08-02 07:23:44 --> File: config.php successfully written
DEBUG - 2024-08-02 07:23:44 --> Start writing database.php
DEBUG - 2024-08-02 07:23:46 --> File: database.php successfully written
DEBUG - 2024-08-02 07:23:46 --> Start creating database structure with assets/install.sql
DEBUG - 2024-08-02 07:23:52 --> Tables successfully created
DEBUG - 2024-08-02 07:23:52 --> Start migrating database to the newest version.
               
```

Secondly I used our original docker-compose file with the exposed http port at 8086 but got the same problem there. I found nothing within the discussions or the bugs but maybe I'm overlooking something - am I doing something wrong, is this a known behaviour (with a known fix)?

The log is identical to the log above (except the timestamps of couse).

With best regards
Patrick

**Comment by HB9HIL** [comment, marked as answer]
Hello Patrick

You could try the following to may get more info:

- Delete the docker volumes to get a clean new install.
- Add another volume for logs `wavelog-logs:/var/www/html/application/logs
- Go through the installer again but this time set in the second Tab "Configuration" > Advanced Settings > Error Log to Level 2 (Debug)
- Run the installer again and then check the logfile which should be available in the wavelog-logs folder (ususally `/var/lib/docker/volumes/...`)


The issue in your case could be: 

During the migration process there is a lockfile which locks the migrator during this process. The fact that your proxy cause a timeout may result that the lockfile did not get deleted. When running the installer again the migration fails in the background because it's still locked (if you did not repull the image). We know about this issue and try to catch more cases where the lockfile not get's deleted. 

However on slow hardware the migration and the DXCC update can take quite a while which can causes issue when running Wavelog behind a Reverse Proxy.

Try again and feedback please

**Reply by winnieXY** [reply]
The hardware shouldn't be too slow. This is a machine with 64G ram and 16 Cores. of which 2 cores and 3G ram are exclusive for that container. 

However your hint on the stale lock file was absolutely right. After removing the the container and recreating it it works now.
